src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient_.cs

```diff
@@ -51,64 +207,49 @@ internal partial class ImageStoreClient
         /// <returns>
         /// A task that represents the asynchronous operation.
         /// </returns>
-        internal Task UploadFileAsync(
+        internal async Task UploadFileAsync(
             byte[] fileContentsToUpload,
             string pathInImageStore,
             string requestId = null,
             long? serverTimeout = 60,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            pathInImageStore.ThrowIfNull(nameof(fileContentsToUpload));
-            pathInImageStore.ThrowIfNull(nameof(pathInImageStore));
-            serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
-            var url = "ImageStore/{pathInImageStore}";
-            url = url.Replace("{pathInImageStore}", Uri.EscapeDataString(pathInImageStore.ToString()));
-            requestId = requestId ?? Guid.NewGuid().ToString();
-            var queryParams = new List<string>();
+            var store = await GetImageStoreConnectionString();
+            if (IsLocalStore(store))
```

Comment:
>Local [](start = 18, length = 5)

This doesn't need to be supported for Local,

Reply:
Why not support it if its part of the interface? If it's an internal method, why is it in the interface?

Reply:
I think we can support uploading the File, but for chunk we should throw IKnvalidOperationException

---
In reply to: [193549620](https://github.com/Microsoft/service-fabric-client-dotnet/pull/12#discussion_r193549620) [](ancestors = 193549620)